plugins/alerting/pkg/alerting/endpoint_backend.go

```diff
@@ -0,0 +1,245 @@
+	err := c.Patch(
+		ctx,
+		dep,
+		patchObj,
+		&client.PatchOptions{
+			FieldManager: "kubectl-rollour",
+		},
+	)
+	return err
+}
+
+type RuntimeEndpointBackend interface {
+	Fetch(ctx context.Context, p *Plugin, key string) (string, error)
+	Put(ctx context.Context, p *Plugin, key string, data string) error
+	Reload(ctx context.Context, p *Plugin) error
+}
+
+// implements alerting.RuntimeEndpointBackend
+type LocalEndpointBackend struct {
+	configFilePath string
+}
+
+func (b *LocalEndpointBackend) Fetch(
+	ctx context.Context, p *Plugin, key string) (string, error) {
+	data, err := os.ReadFile(b.configFilePath)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (b *LocalEndpointBackend) Put(
+	ctx context.Context, p *Plugin, key string, data string) error {
+	err := os.WriteFile(b.configFilePath, []byte(data), 0644)
+	return err
+}
+
+func (b *LocalEndpointBackend) Reload(ctx context.Context,
+	p *Plugin) error {
+	return nil
+}
+
+type K8sEndpointBackend struct {
+	client client.Client
+}
+
+func (b *K8sEndpointBackend) Fetch(
+	ctx context.Context, p *Plugin, key string) (string, error) {
+	name := p.alertingOptions.Get().ConfigMap
+	cfgMap := &corev1.ConfigMap{}
+	err := b.client.Get(ctx, client.ObjectKey{
+		Namespace: "", // check in all?
+		Name:      name,
+	}, cfgMap)
+
+	if err != nil || cfgMap == nil {
+		returnErr := shared.WithInternalServerError(
+			fmt.Sprintf("K8s runtime error, config map : %s not found : %s",
+				name,
+				err),
+		)
+		return "", returnErr
+	}
+
+	if _, ok := cfgMap.Data[key]; !ok {
+		return "", shared.WithInternalServerError(
+			fmt.Sprintf(
+				"K8s runtime error, config map : %s key : %s not found",
+				name,
+				key,
+			),
+		)
+	}
+	return cfgMap.Data[key], nil
+}
+
+func (b *K8sEndpointBackend) Put(ctx context.Context, p *Plugin, key string, data string) error {
+	name := p.alertingOptions.Get().ConfigMap
+	cfgMap := &corev1.ConfigMap{}
+	err := b.client.Get(ctx, client.ObjectKey{
+		Namespace: "", // check in all?
+		Name:      name,
+	}, cfgMap)
+
+	if err != nil || cfgMap == nil {
+		returnErr := shared.WithInternalServerError(
+			fmt.Sprintf("K8s runtime error, config map : %s not found : %s",
+				name,
+				err),
+		)
+		return returnErr
+	}
+	cfgMap.Data[key] = data
+
+	err = b.client.Update(ctx, cfgMap)
+	if err != nil {
+		return shared.WithInternalServerError(
+			fmt.Sprintf("Failed to update alertmanager configmap %s: %s", name, err),
+		)
+	}
+
+	return nil
+}
+
+func (b *K8sEndpointBackend) Reload(ctx context.Context, p *Plugin) error {
+	name := p.alertingOptions.Get().StatefulSet
+	namespace := p.alertingOptions.Get().Namespace
+	statefulSet := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	err := doRolloutRestart(ctx, b.client, statefulSet)
+	if err != nil {
+		return shared.WithInternalServerError(
+			fmt.Sprintf("K8s runtime error, statefulset : %s restart failed : %s",
```

Comment:
For style consistency, don't add spaces before colons here

Reply:
Done